services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/DirectionsResponse.java

```diff
@@ -146,6 +154,49 @@ public static DirectionsResponse fromJson(@NonNull String json) {
     return gson.create().fromJson(json, DirectionsResponse.class).toBuilder().build();
   }
 
+  public static DirectionsResponse fromJson(@NonNull ByteBuffer json, Charset charsets) {
```

Comment:
Thinking about ways we could access specific parts of the DirectionsResponse, rather than deserializing the entire object. The DirectionsResponse can be large, but if we could stream the components.

For example, could we stream the geometry? https://github.com/mapbox/mapbox-java/pull/1518

Reply:
@kmadsen , I'm not sure if I understood this idea correctly. Is the idea about keeping raw json response in memory and deserialise only during first access?